docs/fsharp/style-guide/formatting.md

```diff
@@ -1263,100 +1273,203 @@ module A2 =
+## Formatting types and type annotations
+
+### For types, prefer prefix syntax for generics (`Foo<T>`), with some specific exceptions
+
+F# allows both postfix style of writing generic types (for example, `int list`) as well as the prefix style (for example, `list<int>`).
+Postfix style can only be used with a single type argument.
+Always prefer the .NET style, except for five specific types:
+
+1. For F# Lists, use the postfix form: `int list` rather than `list<int>`.
+2. For F# Options, use the postfix form: `int option` rather than `option<int>`.
+3. For F# Value Options, use the postfix form: `int voption` rather than `voption<int>`.
+4. For F# arrays, use the syntactic name `int[]` rather than `int array` or `array<int>`.
+5. For Reference Cells, use `int ref` rather than `ref<int>` or `Ref<int>`.
+
+For all other types, use the prefix form.
+
+### Formatting function types
 
-Avoid extraneous white space in F# expressions.
+When defining the signature of a function, use white space around the `->` symbol:
 
 ```fsharp
 // OK
-spam (ham.[1])
+type MyFun = int -> int -> string
 
-// Not OK
-spam ( ham.[ 1 ] )
+// Bad
+type MyFunBad = int->int->string
 ```
 
-Named arguments should also not have space surrounding the `=`:
+### Formatting value and argument type annotations
+
+When defining values or arguments with type annotations, use white space after the `:` symbol, but not before:
 
 ```fsharp
 // OK
-let makeStreamReader x = new System.IO.StreamReader(path=x)
+let complexFunction (a: int) (b: int) c = a + b + c
 
-// Not OK
-let makeStreamReader x = new System.IO.StreamReader(path = x)
+let simpleValue: int = 0 // Type annotation for let-bound value
+
+type C() =
+    member _.Property: int = 1
+
+// Bad
+let complexFunctionPoorlyAnnotated (a :int) (b :int) (c:int) = a + b + c
+let simpleValuePoorlyAnnotated1:int = 1
+let simpleValuePoorlyAnnotated2 :int = 2
 ```
 
-### Formatting constructors, static members, and member invocations
+### Formatting return type annotations
 
-If the expression is short, separate arguments with spaces and keep it in one line.
+In function or member return type annotations, use white space before and after the `:` symbol:
 
 ```fsharp
-let person = new Person(a1, a2)
+// OK
+let myFun (a: decimal) b c : decimal = a + b + c // Type annotation for the return type of a function
 
-let myRegexMatch = Regex.Match(input, regex)
+let anotherFun (arg: int) : unit = () // Type annotation for return type of a function
 
-let untypedRes = checker.ParseFile(file, source, opts)
+type C() =
+    member _.SomeMethod(x: int) : int = 1 // Type annotation for return type of a member
+
+// Bad
+let myFunBad (a: decimal) b c:decimal = a + b + c
+
+let anotherFunBad (arg: int): unit = ()
+
+type C() =
+    member _.SomeMethodBad(x: int): int = 1
 ```
 
-If the expression is long, use newlines and indent one scope, rather than indenting to the bracket.
+## Formatting types in signatures
```

Comment:
Should this be H3 since it falls under broader "Formatting Types and Type Annotations" section?